src/pybind/mgr/cephadm/tests/test_services.py

```diff
@@ -223,3 +227,46 @@ def test_iscsi_dashboard_config(self, mock_resolve_ip):
                                        'https://user:password@[FEDC:BA98:7654:3210:FEDC:BA98:7654:3210]:5000')
 
         assert dashboard_expected_call in self.mgr.check_mon_command.mock_calls
+
+
+class TestGrafanaService:
+    @mock.patch("cephadm.serve.CephadmServe._run_cephadm", _run_cephadm('{}'))
+    def testgenerate_config(self, cephadm_module: CephadmOrchestrator):
+        with with_host(cephadm_module, 'test'):
+            with with_service(cephadm_module, GrafanaSpec(initial_admin_password='secure')):
```

Comment:
You can have multiple context managers in a single with statement:
```suggestion
        with with_host(cephadm_module, 'test'), with_service(cephadm_module, GrafanaSpec(initial_admin_password='secure')):
```

Reply:
Right. This is super helpful, if you have lots of mock patches for example:

https://github.com/ceph/ceph/blob/ade690b549c57d0cab8903227bdf984c3772b774/src/cephadm/tests/fixtures.py#L50-L53

Reply:
For that you could use `patch.multiple(cephamd.CephadmDaemon, daemon_path=_daemon_path(), can_run=Mock(return_value=True), ...)`